File: ShanHeScroll_backend/src/main/java/com/mytravel/album/Album.java
package com.mytravel.album;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "album")
@Getter
@Setter
public class Album {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /** 所属用户 */
    @Column(nullable = false)
    private Long userId;

    /** 关联地区（城市/区） */
    @Column(nullable = false)
    private Long regionId;

    /** 相册标题 */
    @Column(nullable = false, length = 100)
    private String title;

    /** 相册描述 */
    @Column(columnDefinition = "TEXT")
    private String description;

    /** 封面图片 URL */
    @Column(length = 500)
    private String coverImage;

    /** 可见性: 0=所有人可见  1=仅自己可见  2=白名单可见  3=黑名单不可见 */
    @Column(nullable = false)
    private Integer visibility = 0;

    @Column(updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    /** 相册内的照片列表（不持久化，由 Service 填充） */
    @Transient
    private List<AlbumPhoto> photos;

    /** 照片数量（不持久化，列表查询时填充） */
    @Transient
    private Long photoCount;

    /** 可见性用户列表（不持久化，查询隐私设置时填充） */
    @Transient
    private List<AlbumVisibilityUser> visibilityUsers;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: ShanHeScroll_front/src/api/album.ts
import http, { extractData } from './index'

export interface AlbumPhoto {
  id: number
  albumId: number
  url: string
  description: string | null
  sortOrder: number
  createdAt: string
}

export interface Album {
  id: number
  userId: number
  regionId: number | null
  title: string
  description: string | null
  coverImage: string | null
  visibility: number
  createdAt: string
  updatedAt: string
  photos: AlbumPhoto[]
  photoCount: number | null
}

/** 我的相册列表 */
export function getMyAlbums() {
  return http.get<Album[]>('/album').then(extractData)
}

/** 创建相册 */
export function createAlbum(data: {
  regionId: number | null
  title: string
  description?: string
  coverImage?: string
}) {
  return http.post<Album>('/album', data).then(extractData)
}

/** 相册详情（含全部照片） */
export function getAlbumDetail(id: number) {
  return http.get<Album>(`/album/${id}`).then(extractData)
}

/** 添加照片 */
export function addPhoto(albumId: number, url: string, description?: string) {
  return http.post(`/album/${albumId}/photo`, { url, description }).then(extractData)
}

/** 更新照片描述 */
export function updatePhotoDesc(albumId: number, photoId: number, description: string) {
  return http.put(`/album/${albumId}/photo/${photoId}`, { description }).then(extractData)
}

/** 删除照片 */
export function deletePhoto(albumId: number, photoId: number) {
  return http.delete<string>(`/album/${albumId}/photo/${photoId}`).then(extractData)
}

/** 删除相册 */
export function deleteAlbum(id: number) {
  return http.delete<string>(`/album/${id}`).then(extractData)
}
